acceptance/tests/agent/enforce_runtimeout.rb

```diff
@@ -0,0 +1,59 @@
+require 'beaker/host_prebuilt_steps'
+
+test_name 'enforce runtimeout' do
+  extend Beaker::HostPrebuiltSteps
+
+  teardown do
+    stop_agent_on(agents)
+    on(master, 'rm -f /etc/puppetlabs/code/environments/production/manifests/site.pp')
+  end
+
+  step 'set up puppet code for putting agents to sleep' do
+    copy_file_to_remote(master, '/etc/puppetlabs/code/environments/production/manifests/site.pp', <<-EOF)
+node default {
+  notify{'Putting agent to sleep': }
+
+  exec{'ruby -e "sleep 60"':
+    path => $facts['path'],
+  }
+}
+EOF
+
+    on(master, 'chmod 0644 /etc/puppetlabs/code/environments/production/manifests/site.pp')
+  end
+
+  step 'test that puppet agent -t kills sleeping daemons' do
```

Comment:
If I'm reading this test correctly its doing what this says test heading says. We should also test that an agent run fails to complete, on its own, when its run takes longer than the time out. We probably want to see this fail both as a daemon (run fails, daemon still is up) and as a onetime run.

Reply:
This test may change with the use of Timeout as both daemonized and non-daemonized runs would be able to time themselves out. I.E. we wouldn't need to spawn a second run to handle the non-daemonized case. Might even be able to remove the acceptance test and handle things up in the unit testing layer.